src/main/java/org/spdx/sbom/gradle/SpdxSbomPlugin.java

```diff
@@ -130,16 +122,9 @@ private void createTaskForTarget(
                   t.usesService(knownLicenseServiceProvider);
 
                   List<String> configurationNames = target.getConfigurations().get();
+                  var rootComponentsProperty =
+                      project.getObjects().listProperty(ResolvedComponentResult.class);
                   for (var configurationName : configurationNames) {
-                    Provider<Set<ResolvedArtifactResult>> artifacts =
```

Comment:
Instead of inferring the artifact from the pom resolution, why not just keep this section in?

Reply:
I believe that's what causing the attribute mismatch errors

Reply:
Correct me if I'm wrong, but I thought that was the pom resolver stuff, while the artifact resolver was actually fine? I'm suggesting we still keep the code in here for dealing with POMs, and for the actual artifacts themselves keep the other section?

Reply:
I'll try again and see what happens

Reply:
Although I'd think that it would be simpler to derive it from the POM vs resolving the artifact.

Reply:
Yeah I guess it doesn't matter too much. It feels like actual vs implied and I would prefer to lean actual. But maybe I'm being difficult and there is no difference.

Reply:
I'm curious how Gradle does it. I'd imagine they have to derive it from the POM as well, because otherwise how would they know.

Reply:
Those artifacts would have already been resolved anyway? We wouldn't be doing extra work, would we?

Reply:
I think there could be some scenarios where they're not (e.g. a CI task that only generates an SBOM, etc...). Although I'm not sure what happens when you do an ARQ, i.e. maybe Gradle will resolve everything anyways instead of just the requested artifact (which would be odd, but who knows)?

Reply:
[Looks like it is likely that Gradle handles it the same way](https://gradle-community.slack.com/archives/CAHSN3LDN/p1700088367801479?thread_ts=1699854619.838439&cid=CAHSN3LDN)

Reply:
Interesting. Are we losing variant information though, by using this method? Would we be presenting the wrong artifact if a variant requested? (Forgive my naivety surrounding variants)

Reply:
I'm not that well versed in it either. I'm not fully clear after reading [the docs](https://docs.gradle.org/current/userguide/variant_model.html) so I can do some investigation with my project and see what happens.

Reply:
Okay just for posterity, conversation continued here: https://gradle-community.slack.com/archives/CAHSN3LDN/p1699854619838439

Reply:
Based on that conversation I'll investigate some alternatives.

Reply:
> Instead of inferring the artifact from the pom resolution, why not just keep this section in?

I tested this and it has issues resolving variants.

Reply:
I think it comes up with Kotlin Multiplatform dependencies. I'll try to get a failing test on main